alpaca/rest.go

```diff
@@ -772,6 +779,126 @@ func (c *client) GetAnnouncement(announcementID string) (*Announcement, error) {
 	return announcement, nil
 }
 
+func (c *client) GetWatchlists() ([]Watchlist, error) {
+	u, err := url.Parse(fmt.Sprintf("%s/%s/watchlists", c.opts.BaseURL, apiVersion))
+	if err != nil {
+		return nil, err
+	}
+
+	resp, err := c.get(u)
+	if err != nil {
+		return nil, err
+	}
+
+	var watchlists []Watchlist
+
+	if err = unmarshal(resp, &watchlists); err != nil {
+		return nil, err
+	}
+
+	return watchlists, nil
+}
+
+func (c *client) CreateWatchlist(req CreateWatchlistRequest) (*Watchlist, error) {
+	u, err := url.Parse(fmt.Sprintf("%s/%s/watchlists", c.opts.BaseURL, apiVersion))
+	if err != nil {
+		return nil, err
+	}
+
+	resp, err := c.post(u, req)
+	if err != nil {
+		return nil, err
+	}
+
+	watchlist := &Watchlist{}
+
+	if err = unmarshal(resp, watchlist); err != nil {
+		return nil, err
+	}
+
+	return watchlist, nil
+}
+
+func (c *client) GetWatchlist(watchlistID string) (*Watchlist, error) {
+	u, err := url.Parse(fmt.Sprintf("%s/%s/watchlists/%s", c.opts.BaseURL, apiVersion, watchlistID))
+	if err != nil {
+		return nil, err
+	}
+
+	resp, err := c.get(u)
+	if err != nil {
+		return nil, err
+	}
+
+	watchlist := &Watchlist{}
+
+	if err = unmarshal(resp, watchlist); err != nil {
+		return nil, err
+	}
+
+	return watchlist, nil
+}
+
+func (c *client) UpdateWatchlist(watchlistID string, req UpdateWatchlistRequest) (*Watchlist, error) {
+	u, err := url.Parse(fmt.Sprintf("%s/%s/watchlists/%s", c.opts.BaseURL, apiVersion, watchlistID))
+	if err != nil {
+		return nil, err
+	}
+
+	resp, err := c.put(u, req)
+	if err != nil {
+		return nil, err
+	}
+
+	watchlist := &Watchlist{}
+
+	if err = unmarshal(resp, watchlist); err != nil {
+		return nil, err
+	}
+
+	return watchlist, nil
+}
+
+func (c *client) AddAssetToWatchlist(watchlistID string, req AddAssetToWatchlistRequest) (*Watchlist, error) {
+	u, err := url.Parse(fmt.Sprintf("%s/%s/watchlists/%s", c.opts.BaseURL, apiVersion, watchlistID))
+	if err != nil {
+		return nil, err
+	}
+
+	resp, err := c.post(u, req)
+	if err != nil {
+		return nil, err
+	}
+
+	watchlist := &Watchlist{}
+
+	if err = unmarshal(resp, watchlist); err != nil {
+		return nil, err
+	}
+
+	return watchlist, nil
+}
+
+func (c *client) RemoveSymbolFromWatchlist(watchlistID string, symbol string) error {
```

Comment:
Any reason why `AddAssetToWatchlist` and `RemoveSymbolFromWatchlist` are asymmetrical? One uses the word "asset" and a request, the other uses "symbol", as an argument. I would pick one. The second is easier to read and use, but the first is more expandable, if any other parameters are ever added to the endpoint.

Reply:
I was following the alpaca documentation Subheading of the api endpoint. :) Now I have replaced `AddAssetToWatchlist` with `AddSymbolToWatchlist` now in the latest commit :)

Reply:
Sorry about being nit-picky here, but now the `AddSymbolToWatchlist` uses `AddSymbolToWatchlistRequest` while `RemoveSymbolFromWatchlist` simply uses the `symbol` as a parameter. Please unify them.

Reply:
Sure. I've fixed it now in the latest commit 😄